samples/server/main.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+
+	"go.opencensus.io/trace"
+	"gocloud.dev/gcp"
+	"gocloud.dev/server"
+	"gocloud.dev/server/sdserver"
+)
+
+type GlobalMonitoredResource struct {
+	projectId string
+}
+
+func (g GlobalMonitoredResource) MonitoredResource() (string, map[string]string) {
+	return "global", map[string]string{"project_id": g.projectId}
+}
+
+func main() {
+	addr := flag.String("listen", ":8080", "HTTP port to listen on")
+
+	ctx := context.Background()
+	credentials, err := gcp.DefaultCredentials(ctx)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+	tokenSource := gcp.CredentialsTokenSource(credentials)
+
+	projectId, err := gcp.DefaultProjectID(credentials)
+	fmt.Printf("projectId = %s\n", projectId)
+	if err != nil {
+		log.Fatal(err)
+	}
+	mr := GlobalMonitoredResource{projectId: string(projectId)}
+	exporter, _, err := sdserver.NewExporter(projectId, tokenSource, mr)
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/hello", func(w http.ResponseWriter, req *http.Request) {
+		fmt.Fprintf(w, "Hello\n")
```

Comment:
I think it would be good to have the handler do something somewhat interesting with one of the APIs to show setting it up, adding health checks, etc.

Reply:
Do you mean something like creating blobs? Sure, we could do that. I just thought the goal here is to demonstrate creating HTTP servers using our server wrappers. Indeed, it was a good goal because it turned out to be non-trivial. ISTM that from here a user shouldn't have much trouble taking code from other samples to add simple blob-management here.

Health checks is an interesting idea - I'll be happy to hear a recommendation of show-casing this in a simple way here

Reply:
If you want to avoid using a portable API, perhaps showing a custom health check that becomes healthy after a few seconds? Something like:

```go
type customHealthCheck struct {
  mu sync.RWMutex
  healthy bool
}

func (h *customHealthCheck) CheckHealth() error {
  h.mu.RLock()
  defer h.mu.RUnlock()
  if !h.healthy {
    return errors.New("not ready yet!")
  }
  return nil
}

// ...

myCheck := new(customHealthCheck)
time.AfterFunc(10 * time.Second, func() {
  myCheck.mu.Lock()
  defer myCheck.mu.Unlock()
  myCheck.healthy = true
})
```

Reply:
Done